plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/AmazonS3Fixture.java

```diff
@@ -312,6 +350,43 @@ public static void main(final String[] args) throws Exception {
             return newInternalError(request.getId(), "Something is wrong with this POST multiple deletes request");
         });
 
+        // non-authorized requests
+
+        TriFunction<String, String, String, Response> credentialResponseFunction = (prefix, key, token) -> {
+            final Date expiration = new Date(new Date().getTime() + TimeUnit.DAYS.toMillis(1));
+            final String response = "{"
+                 + "\"AccessKeyId\": \"" + key + "\","
+                 + "\"Expiration\": \"" + DateUtils.formatISO8601Date(expiration) + "\","
+                 + "\"RoleArn\": \"" + prefix + "_ROLE" + "\","
+                 + "\"SecretAccessKey\": \"" + prefix + "_SCR_KEY" + "\","
+                 + "\"Token\": \"" + token + "\""
+                 + "}";
+
+            final Map<String, String> headers = new HashMap<>(contentType("application/json"));
+            return new Response(RestStatus.OK.getStatus(), headers, response.getBytes(UTF_8));
+        };
+
+        // GET
+        //
+        // http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html
+        handlers.insert(NON_AUTH + " GET /latest/meta-data/iam/security-credentials/", (request) -> {
+            final String response = ec2Key;
```

Comment:
Why `ec2Key` here? This should be the instance profile name, and can reasonably be a fixed value...

Reply:
agree - it does not required to be `ec2Key` (I noticed that in following PR https://github.com/elastic/elasticsearch/pull/31864) - but in fact it requires some instance profile name (it is not required to be fixed as it is 2 step process - to call `/latest/meta-data/iam/security-credentials/` to get the instance profile name and `/latest/meta-data/iam/security-credentials/{instance_profile_name}`)